Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,362 @@
+Reference), this page gives an overview, what can still be used, what is deprecated,
+removed or discouraged and in which way the TSFE class itself may be
+interesting to developers.
+
+Acccessing TSFE
+===============
+
+.. note::
+
+    Some of the former public properties and methods have been changed to
+    protected or marked as internal. You may still find code examples online
+    where :php:`$GLOBALS['TSFE']` is used. In several (but not all) cases, there
+    are alternatives available.
+
+From the source:
+
+    When calling a Frontend page, an instance of this object is available
+    as :php:`$GLOBALS['TSFE']`, even though the core development strives to get
+    rid of this in the future.
+
+:php:`$GLOBALS['TSFE']` is not available in all contexts. In particular, it is
+only available in frontend contexts, not in the backend or cli.
+
+Initializing :php:`$GLOBALS['TSFE']` in the Backend is sometimes done in code
+examples found online. TSFE is not initialized in the backend context by the
+core (and there is usually no need to do this).
+
+From the PHP documentation:
+
+    As of PHP 8.1.0, $GLOBALS is now a read-only copy of the global symbol table.
+    That is, global variables cannot be modified via its copy.
+
+https://www.php.net/manual/en/reserved.variables.globals.php
+
+Howtos
+======
+
+.. add howtos for common use cases, remove the not recommended, deprecated
+.. way if there is a better way
+
+Following are some examples which use TSFE and alternatives to using TSFE,
+where available:
+
+-------------
+
+Access the :php:`ContentObjectRenderer` (often referred to as "cObj"):
+
+.. code-block:: php
+
+    $GLOBALS['TSFE']->cObj
+
+Can be done as follows in Extbase controllers:
+
+.. code-block:: php
+
+    $contentObj = $this->configurationManager->getContentObject();
+
+-------------
+
+Access the current page id:
+
+.. code-block:: php
+
+    $GLOBALS['TSFE']->id
+
+Can be done using the ContentObjectRenderer (see previous example):
+
+.. code-block:: php
+
+    $pageId = $contentObj->data['pid'];
+
+-------------
+
+Access language settings is now done using the
+:ref:`language aspect <context_api_aspects_language>`.
+
+Get the request language of the current page as integer:
+
+.. code-block:: php
+
+    $context->getPropertyFromAspect('language', 'id');
+
+--------------
+
+Accessing information about Frontend and Backend users.
+
+.. code-block:: php
+
+    $feUser = $GLOBALS['TSFE']->fe_user;
+
+Some information via frontend and backend users con be obtained via the
+:ref: `user aspect <context_api_aspects_user>`. For example:
+
+.. code-block:: php
+
+    // return whether a frontend user is logged in
+    $context->getPropertyFromAspect('frontend.user', 'isLoggedIn');
+
+------------------
+
+To get the base URL of the current site, use site configuration.
+
+For example, to get the site via the current page id:
+
+.. code-block:: php
+
+    $siteFinder = GeneralUtility::makeInstance(SiteFinder::class);
```

Comment:
One should use either dependency injection to get the SiteFinder or just use the site request attribute: https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/Site.html

Background: We would like to get rid of GeneralUtility::makeInstance() and use DI as best practise wherever possible.

Reply:
done. Thanks for the link.